src/views/catalog/utils/WizardTemplateContext.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+
+import { getTemplateVirtualMachineObject } from './vm-template-source/utils';
+
+type WizardTemplateContextType = {
+  /** the template used for the wizard */
+  template?: V1Template;
+  /**
+   * updates the template vm object. The provided vm object will be merged with the existing vm object
+   * @param vmObject V1VirtualMachine
+   */
+  updateVmObject?: (vmObject: V1VirtualMachine) => void;
```

Comment:
why not just provide the new template  ?

looks more correct to me to have a template object in the session state
and then replace it each time it changes

e.g.
```
session state:
{
    templae-<name>-<namespace>-<uid>: the-template-obj
    templae-<name>-<namespace>-<uid>: template...
}
```
clean after a vm is created, or let it clean up each session